examples/generative/speaker_recognition_using_cnn.py

```diff
@@ -0,0 +1,486 @@
+        self.scale = scale
+
+        self.input_size = self.noise_stream.shape[1] // 2
+        self.input_dim = (self.input_size, 1)
+
+    def __len__(self):
+        return int(np.ceil(len(self.file_paths) / float(self.batch_size)))
+
+    def __getitem__(self, idx):
+        i = idx * self.batch_size
+
+        # Getting the batches of files
+        batch_file_paths = self.file_paths[i : i + self.batch_size]
+
+        # Reading audio file
+        Xt = [wavfile.read(file)[1] for file in batch_file_paths]
+        Xt = np.array(Xt)
+
+        # Adding noise
+        if self.noises is not None:
+            prop = np.max(Xt, axis=-1, keepdims=True) / np.max(
+                self.noise_stream[self.noises[i : i + self.batch_size], :],
+                axis=-1,
+                keepdims=True,
+            )
+            Xt = (
+                Xt
+                + self.noise_stream[self.noises[i : i + self.batch_size], :]
+                * prop
+                * self.scale
+            )
+
+        # Let's now get the Fast Fourier Transformation (FFT)
+        # in order to transform our signal into the frequency domain.
+        # We will take only the positive frequencies hence we are taking
+        # only Xf[:,:self.input_size] where input_size is half the desired sr
+        Xf = np.fft.fft(Xt, axis=-1)
+        Xf = np.expand_dims(np.abs(Xf[:, : self.input_size]), axis=-1)
+
+        # Labels
+        y = self.labels[i : i + self.batch_size]
+
+        return Xf, y
+
+    def on_epoch_end(self):
+        # At the end of each epoch, we shuffle the noises so that
+        # the same audio sample to be trained with different noise
+        # samples in order to reduce overfitting
+        if self.noises is not None and self.shuffle_noises is True:
+            rng = np.random.RandomState(self.shuffle_seed)
+            rng.shuffle(self.noises)
+
+
+# Training sample generator with noises added to the sample
+train_gen = SpeechGenerator(
+    file_paths=train_file_paths,
+    labels=train_labels,
+    class_names=class_names,
+    noise_stream=noise_stream,
+    noises=train_noises,
+    batch_size=BATCH_SIZE,
+    shuffle_noises=True,
+    shuffle_seed=SHUFFLE_SEED,
+    scale=SCALE,
+)
+
+# Validation sample generator with no noise added
+valid_gen = SpeechGenerator(
+    file_paths=valid_file_paths,
+    labels=valid_labels,
+    class_names=class_names,
+    noise_stream=noise_stream,
+    noises=None,
+    batch_size=32,
+    shuffle_noises=True,
+    shuffle_seed=SHUFFLE_SEED,
+    scale=SCALE,
+)
+
+"""
+## Model Definition
+
+I am using a model that is very similar to the `VGG16` for image classification where I
+replaced the `Conv2D` and `MaxPool2D` by `Conv1D` and `MaxPool1D` respectively.
+"""
+
+keras.backend.clear_session()
+
+np.random.seed(42)
+tf.random.set_seed(42)
+
+
+def build_model(input_shape, num_classes):
+    model = keras.models.Sequential(
```

Comment:
In general, prefer using the Functional API. Your model is pretty deep, so consider adding residual connections. Note that it is generally not necessary to pass the `name` argument to very layer; usually it's only useful for input and output layers.